include/rsp_queue.inc

```diff
@@ -0,0 +1,447 @@
+#ifndef RSP_QUEUE_INC
+#define RSP_QUEUE_INC
+
+#include <rsp.inc>
+#include <rspq_constants.h>
+
+# Globally reserved registers
+#define rspq_dmem_buf_ptr gp
+
+# Can be used with l* instructions to get contents of the current command at the specified offset.
+# The total command size needs to be specified as well.
+#define CMD_ADDR(offset, cmdsize) (%lo(RSPQ_DMEM_BUFFER) + (offset) - (cmdsize)) (rspq_dmem_buf_ptr)
+
+#define OVERLAY_HEADER_SIZE 0x8
+
+.macro RSPQ_BeginOverlayHeader stateStart, stateEnd
```

Comment:
It would be great to have this documented as well. (Sorry if I was unable to find it)

Reply:
I realized that rsp_queue.inc and writing overlays wasn't documented very well in general, so I added a bunch of docs. Also I changed the API a little bit to make it less error prone and a little bit nicer as well. What do you think?